src/openforms/js/components/admin/form_design/form-creation-form.js

```diff
@@ -534,6 +533,35 @@ function reducer(draft, action) {
       draft.logicRulesToDelete = [];
       break;
     }
+    /**
+     * Form Variables
+     */
+    case 'ADD_USER_DEFINED_VARIABLE': {
+      draft.formVariables.push(EMPTY_VARIABLE);
+      break;
+    }
+    case 'DELETE_USER_DEFINED_VARIABLE': {
+      const key = action.payload;
+      draft.formVariables = draft.formVariables.filter(variable => variable.key !== key);
+      break;
+    }
+    case 'CHANGE_USER_DEFINED_VARIABLE': {
+      const {key, propertyName, propertyValue} = action.payload;
+
+      const index = draft.formVariables.findIndex(variable => variable.key === key);
+      const existingKeys = draft.formVariables.map(variable => variable.key);
+
+      if (propertyName === 'key' && existingKeys.includes(propertyValue)) {
+        draft.formVariables[index]['key'] = getUniqueKey(propertyValue, existingKeys);
+      } else {
+        draft.formVariables[index][propertyName] = propertyValue;
+      }
+      break;
+    }
+    case 'ADD_STATIC_VARIABLES': {
+      draft.formVariables = [...DEFAULT_STATIC_VARIABLES];
+      break;
+    }
```

Comment:
what's the plan here for existing forms - data migration? Since loading the existing form editor doesn't show me any static variables.

Reply:
Yes I was thinking a data migration. Should I add it to this PR?